labs/lab06.md

```diff
@@ -0,0 +1,75 @@
+- [ ] Identify at least two issues (no need to name the anti-pattern) associated with the Drawing System, and explain how you would refactor it. No coding required.
+
+## Instructions
+
+### Setup
+<u>Fork</u> and clone the repo from [https://github.com/CMU-17-214/f23-lab06](https://github.com/CMU-17-214/f23-lab06.git).
+
+The first two tasks are explicitly referring to Java code (the typescript codebase is also provided, but as longer legacy version from previous semesters). The last task is provided both in Java and Typescript, so feel free to explore the one you are most comfortable with. If you would like additional exercise, feel free to look into the "accounts" folder.
+
+### Task 1
+Frogger is trying to cross the road, which she holds as a "Road" object in her fields. The latter holds a boolean array indicating which steps are “occupied”; Frogger is on a specific square (“position”) and provides a move method (either forward or backward).
+
+In the frogger folder, navigate to the Frogger.java and Road.java classes and take a look at the code. What is it doing and what anti-pattern is present? Then try to fix the design by modifying the two classes.
+
+> Hint: 
+> + What information should the Frogger hold? Does it do things that appear weird for a Frogger?
+
+
+### Task 2
+
+Frogger crossed the road and remembered: she was here to *record* her identity status at the Frogger Office. That's why there is a "Records" object in her field. Frogger tries to fill in the fields according to the Record object. But there is simply too many things to fill in--Frogger is annoyed.
+
+Take a look at Records.java and then Frogger.java (again). What anti-pattern is present? Then try to fix the design by modifying the two classes... or, take a look at FroggerID.java. How might you use it for convenience?
+
+### Task 3
+
+Next, let's study the “drawing” system. Open the drawing folder. The system has multiple substantial flaws, some in the same components. Rather than working top-down, trying to retrieve violations of design principles in the code, let’s work bottom up, identifying code smells and anti-patterns, and then strategizing about how to remedy those.
+
+1) Read through every file, making note of anything that stands out. This can be more specific than code smells; think: duplicated code, problematic ‘instanceof’ checking, poor extensibility. While you are reading, try to identify the “control-flow” of this code; who calls whom (e.g.: Shape calls Line), and where would the first user-based call come from? What does the system do?
+
+2) Try to map your concerns to anti-patterns. Is everything that stood out really problematic? Is it all urgently in need of a solution? If not, what would you tackle first? Reason through the design implications of each change. Some will be very minor, some major.
+
+Prepare at least two refactorings.
+Is it as seamless as you thought it would be?
+Do you prefer the code afterwards?
```

Comment:
there's a lot of stuff here. more than we actually require them to do. can you go over all text again and identify what's essential to do the task and what are hints or additional suggestions?

for example for task 3, we do not expect them to execute any refactorings. and (1) and (2) are really hints at how to proceed with the task?